From db219aee1747888d8209cc6c433611407f055af1 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 1 May 2015 19:33:40 +0530
Subject: [PATCH] [typec-detect]: Remove USB_EVENT_VBUS notification from typec
 detection

USB is coming out of D3-state in UFP mode. This is happening as, USB_EVENT_VBUS
notification is sent from detection driver.
To avoid this, BC1.2 charger detection module will send the USB_EVENT_VBUS based
on the charger type.

Change-Id: Ic98e478ad57c5df62d3feef2674c1c8d89b7915a
Tracked-on: https://jira01.devtools.intel.com/browse/IMINAN-34837
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/usb/typec/usb_typec_detect.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index 4a427c4bb81d..1395c5ee19b5 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -368,9 +368,6 @@ static void update_phy_state(struct work_struct *work)
 			typec_setup_cc(phy, use_cc, TYPEC_STATE_ATTACHED_UFP);
 			extcon_set_cable_state(detect->edev, "USB", true);
 
-			/* notify usb */
-			atomic_notifier_call_chain(&detect->otg->notifier,
-					USB_EVENT_VBUS, NULL);
 			/* notify power supply */
 			cable_props.chrg_evt =
 					POWER_SUPPLY_CHARGER_EVENT_CONNECT;
@@ -410,11 +407,10 @@ static void update_phy_state(struct work_struct *work)
 			mutex_unlock(&detect->lock);
 			queue_work(detect->wq_lock_ufp,
 					&detect->lock_ufp_work);
+			atomic_notifier_call_chain(&detect->otg->notifier,
+					USB_EVENT_NONE, NULL);
 			break;
 		}
-		/* setup data mux */
-		atomic_notifier_call_chain(&detect->otg->notifier,
-					USB_EVENT_NONE, NULL);
 		mutex_lock(&detect->lock);
 		detect->state = DETECT_STATE_UNATTACHED_DRP;
 		mutex_unlock(&detect->lock);
-- 
1.9.1

